h2o-core/src/main/java/water/rapids/ast/prims/advmath/SpearmanCorrelation.java

```diff
@@ -1,72 +1,155 @@
 package water.rapids.ast.prims.advmath;
 
+import water.Key;
 import water.MRTask;
 import water.Scope;
 import water.fvec.Chunk;
 import water.fvec.Frame;
 import water.fvec.Vec;
-import water.rapids.Env;
 import water.rapids.Merge;
-import water.rapids.Val;
-import water.rapids.ast.AstPrimitive;
-import water.rapids.ast.AstRoot;
-import water.rapids.vals.ValNum;
 import water.util.FrameUtils;
-import water.util.VecUtils;
 
 import java.math.BigDecimal;
 import java.math.MathContext;
+import java.util.Objects;
+
+public class SpearmanCorrelation {
+
+  public static Frame calculate(final Frame frameX, Frame frameY, final AstCorrelation.Mode mode) {
+    Objects.requireNonNull(frameX);
+    Objects.requireNonNull(frameY);
+    assert frameX.numCols() == frameY.numCols();
```

Comment:
this duplicates the assert in checkCorrelationDoable

Reply:
Removed the one from the check.